### ISSUE TITLE
Bug: i3bar disappeared after aptitude upgrade

### ISSUE THREAD

**Issue by eduncan911** [regolith-linux/regolith-desktop#359]
**OS:** 

Pop!_OS 19.10 (Ubuntu 19.10)

**What is the version of i3 and i3bar?**

```
$ i3 --version
i3 version 4.17.1 (2019-08-30) © 2009 Michael Stapelberg and contributors
$ i3bar --version
i3bar version 4.17.1 (2019-08-30) © 2010 Axel Wagner and contributors
```

**What was I doing:**

    sudo apt update && sudo apt upgrade

**What was I expecting:**

No changes to i3bar, or really anything with regolith.

**What did happen:**

Everything was working for a few days, until I restarted/reloaded i3 config. then i3bar disappeared (and theming looks different). 

**What did you try to resolve it?**

* Rebooting
* Restarting apps
* Reloading i3-gap
* Rebooting again
* Re-installed regolith, reloaded
* Restarted again

**Have you changed any configs?**

Yes, however I renamed the config and restarted i3 - nada.

This is the single only mod I have in my config file:

```
$ cat ~/.config/regolith/Xresources
i3-wm.bar.position: top
```

![Screenshot from 2020-04-30 00-10-46](https://user-images.githubusercontent.com/104980/80671498-06063480-8a78-11ea-93ec-14f5af84d9a5.png)

**Comment by ssauperl** [comment]
I had the same issue when upgrading from 1.3 to 1.4.  I got it fixed with updating i3 config in my home folder:
`cp /etc/regolith/i3/config ~/.config/regolith/i3/config`
Guess is worth a try. If you have any customisation in your config i recommend making a backup of `~/.config/regolith/i3/config`

**Comment by eduncan911** [comment]
Thanks @ssauperl . Unfortunately, that did not work.  

```
$ cp /etc/regolith/i3/config ~/.config/regolith/i3/config
$ ls -la ~/.config/regolith/i3/config
-rw-r--r-- 1 eric eric 26243 Apr 30 07:08 /home/eric/.config/regolith/i3/config
```

Reloaded i3, nada.  Rebooted machine, nada.

Also, [the current docs for `R1.3.1` say not to use these files](https://regolith-linux.org/docs/howto/stage-configs/) any longer but to change to using `Xresources` for overrides.

Updated my OP with the versions of things.

**Comment by eduncan911** [comment]
I'm wondering if I am running an older version of `Regolith-desktop`?

As noted in the OP, I am not running Regolith OS.  Just using the desktop ontop of Ubuntu.

```
$ cat /var/log/dpkg.log | grep regolith-desktop
2020-04-22 20:22:49 install regolith-desktop:amd64 <none> 2.49-1ubuntu1~ppa1
2020-04-22 20:22:49 status half-installed regolith-desktop:amd64 2.49-1ubuntu1~ppa1
2020-04-22 20:22:49 status unpacked regolith-desktop:amd64 2.49-1ubuntu1~ppa1
2020-04-22 20:23:00 configure regolith-desktop:amd64 2.49-1ubuntu1~ppa1 <none>
2020-04-22 20:23:00 status unpacked regolith-desktop:amd64 2.49-1ubuntu1~ppa1
2020-04-22 20:23:00 status half-configured regolith-desktop:amd64 2.49-1ubuntu1~ppa1
2020-04-22 20:23:00 status installed regolith-desktop:amd64 2.49-1ubuntu1~ppa1
2020-04-28 19:12:47 upgrade regolith-desktop:amd64 2.49-1ubuntu1~ppa1 2.58-1eoan1
2020-04-28 19:12:47 status half-configured regolith-desktop:amd64 2.49-1ubuntu1~ppa1
2020-04-28 19:12:47 status unpacked regolith-desktop:amd64 2.49-1ubuntu1~ppa1
2020-04-28 19:12:47 status half-installed regolith-desktop:amd64 2.49-1ubuntu1~ppa1
2020-04-28 19:12:47 status unpacked regolith-desktop:amd64 2.58-1eoan1
2020-04-28 19:15:17 configure regolith-desktop:amd64 2.58-1eoan1 <none>
2020-04-28 19:15:17 status unpacked regolith-desktop:amd64 2.58-1eoan1
2020-04-28 19:15:17 status half-configured regolith-desktop:amd64 2.58-1eoan1
2020-04-28 19:15:17 status installed regolith-desktop:amd64 2.58-1eoan1
2020-04-30 00:09:30 upgrade regolith-desktop:amd64 2.58-1eoan1 2.58-1eoan1
2020-04-30 00:09:30 status half-configured regolith-desktop:amd64 2.58-1eoan1
2020-04-30 00:09:30 status unpacked regolith-desktop:amd64 2.58-1eoan1
2020-04-30 00:09:30 status half-installed regolith-desktop:amd64 2.58-1eoan1
2020-04-30 00:09:30 status unpacked regolith-desktop:amd64 2.58-1eoan1
2020-04-30 00:09:30 configure regolith-desktop:amd64 2.58-1eoan1 <none>
2020-04-30 00:09:30 status unpacked regolith-desktop:amd64 2.58-1eoan1
2020-04-30 00:09:30 status half-configured regolith-desktop:amd64 2.58-1eoan1
2020-04-30 00:09:30 status installed regolith-desktop:amd64 2.58-1eoan1
```

**Comment by kgilmer** [comment]
What does `i3 --moreversion` say and `file ~./Xresources-regolith`?  This section on the website describes how to get to a "vanilla" state from which you can integrate your changes back in: https://regolith-linux.org/docs/getting-started/install/#upgrading-with-customizations

**Comment by kgilmer** [comment]
`2.58-1eoan1` is the current version of the desktop package.  See here: https://launchpad.net/~regolith-linux/+archive/ubuntu/release

**Comment by eduncan911** [comment]
> What does `i3 --moreversion` say and `file ~./Xresources-regolith`? This section on the website describes how to get to a "vanilla" state from which you can integrate your changes back in: https://regolith-linux.org/docs/getting-started/install/#upgrading-with-customizations

Thanks.  I'll try this when I am back on my machine.

But note that I have no customization, outside of the one line i3bar change in the OP.

But even then, I removed that file and nothing changes - except it goes back to the bottom.

**Comment by eduncan911** [comment]
At this point, I have followed the instructions and purged the entire thing from my system.

The idea was to force install the older version, or move to `stable` instead of `release` branch in the PPA.

**Comment by jcswart** [comment]
I have  the same issue. 

One strange behavior I just noticed is that I had an `~/.config/regolith/i3/` folder with nothing it. It was created to make a custom i3 config before I learned that I could put any reasonable value I wanted in `Xresources`.

I was also seeing my own gnome-terminal, installed during regolith `1.3`.

I deleted the empty `~/.config/regolith/i3` folder and now I am getting the new gnome-terminal installed via 1.4. So the existence of paths may trigger configuration options even if the files do not exist.

I'll try the vanilla state instructions tonight.

**Comment by eduncan911** [comment]
Nice.  Yes, I suspected that too - that I had empty path but didn't think of deleting it before.

I'll also try tonight after work.

**Comment by jcswart** [comment]
Following https://regolith-linux.org/docs/getting-started/install/#reinstallation did not fix the issue for me either. 

I only have the Notification Bell and a `0`. No other i3 bar items appear. This was the case previously.

**Comment by jcswart** [comment]
Oh wow, okay I see the issue (or at least my problem).

Under https://regolith-linux.org/docs/getting-started/install/#option-2-ppa you can see the optional steps to install various items via `i3xrocks`. These are the items placed in the area typically referred to as the `i3-bar`.

I would not have guessed the time and calendar, etc. to considered optional. I do not know if I manually installed these when I installed 1.3 or if they came as part of it.

When doing a clean install straight to `1.4`. 

So this is possibly not a bug just an expected assumption that things other than notifications like the time and battery indicator would be defaults.

**Comment by kgilmer** [comment]
> I would not have guessed the time and calendar, etc. to considered optional. I do not know if I manually installed these when I installed 1.3 or if they came as part of it.

Yes, this was a change in 1.4 in order to support arm64 devices and pluggable notification components.  Unfortunately there was no simple way of performing these package upgrades without then causing them to be required, or doing something hackish.  Sorry for the confusion.

If you have any suggestions to make the documentation more clear, please let me know!  :smile:

**Comment by jcswart** [comment]
No the install docs are clear, maybe a note in the upgrades section? Either way this project continues to be awesome! Thanks for the hard work.

**Comment by eduncan911** [comment]
Sorry for not getting back soon.  Life and all.

It would seem the "release" branch I used (cause I just copied and pasted the install instructions) does differ from the "stable" branch in that it's treated more of a beta release channel, before things go to stable?  

That would make sense as the product does work, and as we've found in this thread it's the lack of notification of what was changing with a single update/upgrade that broke it.

To be clear, yes, following the updated documentation to install the additional deps - that didn't exist earlier when I first installed - fixed it.  :)

So, to close this issue, I think this is a great learning opportunity for Regolith support for the next "breaking" change to be announced a bit differently (or at all?).  So a few suggestions and we can then close the issue:

* Arch Linux used to have a (but no longer!) a Twitter account for announcing breaking changes in their rolling releases.  That's it.  Short, clear, "X package needs Y work because Z will break" and a link for more info with a short command to run.  They have switched to an RSS feed now for the same.  Perhaps Regolith can have a "breaking change" similar RSS feed we can subscribe to and get alerts for - before we upgrade.  
* Add upgrade/changeme docs about the breaking changes and resolutions.  However, this doesn't show / notify users too easily especially if every release just has a bunch of feature and bug fixes, where the breaking changes would be hidden way down in the notes (see Arch's dedicated channel notification above).
* Last thought, which I think may be the more feasible and automatic approach: force an aptitude Notes pop up with a required user interaction prompt - e.g. they have to read the break change notes, before aptitude can continue.

That last note... I didn't think was possible with aptitude until the Spectra and Meltdown patches and upgrades of Ubuntu 14.04 and it's old 3.X kernel, needing to force upgrade to 4.X kernel.

But yet, the GRUB package forced that CLI prompt to force the user to decide on what action to take with their bootloader...  

...which, I think was genius.  No automatic breaking, no assumptions made by the package maintainers.  

Some other aptitude packages have also popped up a Notes field to tell users about a breaking change.

Now, these pop-ups can break auto-update scripts.  But at the same time, this is a desktop environment - not a server.  So user interaction is expected.

**Comment by eduncan911** [comment]
Sorry for not getting back soon.  Life and all.

It would seem the "release" branch I used (cause I just copied and pasted the install instructions) does differ from the "stable" branch in that it's treated more of a beta release channel, before things go to stable?

That would make sense as Regolith does work well, and the upgrade went fine - just it was missing any type of notification of the change in the release channel - as we've found in this thread with a single update/upgrade that broke it.

To be clear, yes, following the updated documentation to install the additional deps - that didn't exist earlier when I first installed - fixed it.  :)

Closing this issue as that's the resolution: follow the updated docs and install the new deps that are now modular and required by most installs.

I've also opened #369 to address notifications of breaking changes going forward.  :)
